lib/aoc_rb/app.rb

```diff
@@ -108,12 +108,22 @@ def exec(year = options[:year], day = options[:day])
     method_option :day, aliases: "-d", type: :numeric, default: Time.now.day
 
     def output(year = options[:year], day = options[:day])
-      puzzle = AocRb::PuzzleSource.create_puzzle(year, day)
       input = AocRb::PuzzleInput.load(year, day)
+      puzzle = AocRb::PuzzleSource.create_puzzle(year, day, input)
 
-      AocRb::PuzzleSource.run_part('part 1') { puzzle.part_1(input) }
+      AocRb::PuzzleSource.run_part('part 1') { puzzle.part_1 }
       puts
-      AocRb::PuzzleSource.run_part('part 2') { puzzle.part_2(input) }
+      AocRb::PuzzleSource.run_part('part 2') { puzzle.part_2 }
+    end
+
+    desc "spec", "runs tests for the given day"
```

Comment:
I think it would it make sense for `spec` to run all of the specs. With options for narrowing it down to a specific day? Or something like `spec:today` as a shortcut?

Reply:
I added it this way to maintain consistency across all methods. How do you feel about instead adding a `spec:all` to run all the tests.

Also when you mention all the tests, do you mean for the current year or everything there is in the `spec` folder?

Reply:
I guess that makes sense. Everything else defaults to running the current day. I think it would be nice to be able to run all of them though.

FWIW, I have a GitHub workflow on my AOC project that runs all of my tests when I push any solutions up to the repo, just using `bundle exec rspec`.